client/asset/dcr/wallet.go

```diff
@@ -92,54 +86,64 @@ type Wallet interface {
 	// for non-wallet outputs. Returns asset.CoinNotFoundError if the unspent
 	// output cannot be located.
 	UnspentOutput(ctx context.Context, txHash *chainhash.Hash, index uint32, tree int8) (*TxOutput, error)
-	// GetNewAddressGapPolicy returns an address from the specified account using
-	// the specified gap policy.
-	GetNewAddressGapPolicy(ctx context.Context, account string, gap dcrwallet.GapPolicy) (stdaddr.Address, error)
-	// SignRawTransaction signs the provided transaction.
-	SignRawTransaction(ctx context.Context, txHex string) (*walletjson.SignRawTransactionResult, error)
+	// ExternalAddress returns a new external address,
+	ExternalAddress(ctx context.Context, acctName string) (stdaddr.Address, error)
+	// InternalAddress returns a change address from the Wallet.
+	InternalAddress(ctx context.Context, acctName string) (stdaddr.Address, error)
+	// SignRawTransaction signs the provided transaction. SignRawTransaction
+	// is not used for redemptions, so previous outpoints and scripts should
+	// be known by the wallet.
+	SignRawTransaction(context.Context, *wire.MsgTx) (*wire.MsgTx, error)
```

Comment:
I think this doc should mention that the `msgTx` passed in could be modified, so callers should generally pass in a copy.

Reply:
I'm not seeing any modifications made by `(*rpcWallet).SignRawTransaction` to the input `MsgTx`, so alternatively this interface comment should require the implementations do not modify the input.